app/models/concerns/public_asset.rb

```diff
@@ -24,6 +24,11 @@ def public_url_token(options={})
     n = o[:name]
     x = o[:extension]
 
+    if x.blank? && n.include?('.')
```

Comment:
You were starting to explain this in person before, but I can't remember exactly the reasoning -- wherefore `include?('.')`? are we looking for cases other than n = `filename.` ?